brave/brave-core
# [ads] Optimize database queries serialization/deserialization overhead

Resolves https://github.com/brave/brave-browser/issues/42598
Resolves https://github.com/brave/brave-browser/issues/41632



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Make sure that Brave Ads core functionality works as expected:
- Join Brave Rewards
- Make sure that search Ads, NTT, Brave News, Notification ads can be served, viewed and clicked
- Make sure that served, viewed and clicked confirmations are sent to the server
